modules/user/client-react/components/LoginView.native.jsx

```diff
@@ -20,18 +20,20 @@ class LoginView extends React.PureComponent {
   }
 
   handleOpenURL = async ({ url }) => {
-    // Extract stringified user string out of the URL
-    const [, data] = url.match(/data=([^#]+)/);
-    const decodedData = JSON.parse(decodeURI(data));
-    const { client } = this.props;
-    if (decodedData.tokens) {
-      await setItem('accessToken', decodedData.tokens.accessToken);
-      await setItem('refreshToken', decodedData.tokens.refreshToken);
-    }
-    await client.query({ query: CURRENT_USER_QUERY });
+    if (url.match(/data=([^#]+)/)) {
+      // Extract stringified user string out of the URL
+      const [, data] = url.match(/data=([^#]+)/);
```

Comment:
You can specify the regex as a constant to avoid duplication.

Reply:
Done